src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareIncludeExcludeNodeOperation.cs

```diff
@@ -58,7 +65,45 @@ public void Execute(TaskExecutionMode mode)
                     throw new InvalidOperationException(SR.SchemaCompareExcludeIncludeNodeNotFound);
                 }
 
+                // Check first if the dependencies will allow this if it's an exclude request
+                if (!this.Parameters.IncludeRequest)
+                {
+                    IEnumerable<SchemaDifference> dependencies = this.ComparisonResult.GetExcludeDependencies(node);
+
+                    bool block = false;
```

Comment:
This foreach could be a single linq statement: bool block = dependencies.Any(d => d.Included);

Reply:
♥ Linq